GDJS/Runtime/runtimeobject.js

```diff
@@ -1151,12 +1151,50 @@ gdjs.RuntimeObject.collisionTest = function(obj1, obj2) {
 };
 
 /**
- * Check the distance between two objects.
- * @method distanceTest
+ * @method raycastTest
  * @static
- */
-gdjs.RuntimeObject.distanceTest = function(obj1, obj2, distance) {
-    return obj1.getSqDistanceToObject(obj2) <= distance;
+ * @param x {Number} The raycast source X
+ * @param y {Number} The raycast source Y
+ * @param angle {Number} The raycast angle
+ * @param dist {Number} The raycast max distance
+ * @param closest {Boolean} Get the closest or farthest collision mask result?
+ * @return A raycast result with the contact points and distances
+ */
+gdjs.RuntimeObject.prototype.raycastTest = function(x, y, angle, dist, closest) {
+    var objW = this.getWidth();
+    var objH = this.getHeight();
+    var diffX = this.getDrawableX()+this.getCenterX() - x;
+    var diffY = this.getDrawableY()+this.getCenterY() - y;
+    var boundingRadius = Math.sqrt(objW*objW + objH*objH)/2.0;
+
+    var result = gdjs.Polygon.raycastTest._statics.result;
+    result.collision = false;
+
+    if ( Math.sqrt(diffX*diffX + diffY*diffY) > boundingRadius + dist )
+        return result;
+
+    
+    var endX = x + dist*Math.cos(angle*3.14159/180.0);
+    var endY = y + dist*Math.sin(angle*3.14159/180.0);
+    // var testSqDist = closest ? 0 : Number.MAX_VALUE;
```

Comment:
You can remove this comment ;)

Reply:
I leave lot of debug things out there :facepalm: